cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -748,32 +748,35 @@ def _add_efs_mount_target(
             checked_availability_zones.append(availability_zone)
 
     def _add_raid_volume(self, id: str, shared_ebs: SharedEbs):
-        """Add specific Cfn Resources to map the EBS storage."""
-        # TODO: add RAID storage
-        return f"fake_id{id}-{shared_ebs.mount_dir}"
+        """Add specific Cfn Resources to map the RAID EBS storage."""
+        ebs_ids = []
+        for _ in range(shared_ebs.raid.number_of_volumes):
+            ebs_id = self._add_cfn_volume(id, shared_ebs).ref
+            ebs_ids.append(ebs_id)
+        return ebs_ids
```

Comment:
You should change the code calling _add_raid_volume because it's expecting a single ebs id.
Or let's move the `for _ in range(shared_ebs.raid.number_of_volumes):` loop in the _add_shared_storage

Reply:
For EBS, EFS, and FSx, each item of the list represents a whole entity of EBS, EFS, and FSx. However, for RAID, if we don't use "lists in a list", each item of the list only represents a part of the RAID. And what shall we do if we want to support multi RAIDs?

Reply:
We'll figure out when adding the support for multi RAIDs. :D
For now we just need the list of ids, that will be used as `"cfn_raid_vol_ids": "${RAIDVolIds}",`
so we can use the same approach of the ebs one.

Reply:
Done